code-name-taran/src/com/spaghettisoft/component/sixtysyx/resourses/Deck.java

```diff
@@ -0,0 +1,81 @@
+package com.spaghettisoft.component.sixtysyx.resourses;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.spaghettisoft.component.sixtysyx.resourses.constants.Colour;
+import com.spaghettisoft.component.sixtysyx.resourses.constants.Rank;
+
+public class Deck
+{
+	Queue<PlayingCard> deck;
+	public final int DECK_SIZE = 24;
+
+	public Deck()
+	{
+		newDeck();
+	}
+
+	private void newDeck()
+	{
+		deck = new LinkedList<>();
+		generateDeck();
+	}
+
+	private void shuffle(PlayingCard[] deckCopy)
+	{
+		Random rnd = ThreadLocalRandom.current();
+		
+		for (int i = deckCopy.length-1; i > 0; i--)
+		{
+			int index = rnd.nextInt(i+1);
+			PlayingCard temp = deckCopy[index];
+			deckCopy[index] = deckCopy[i];
+			deckCopy[i] = temp;
+		}
+		arrayToQueue(deckCopy);
+		
+
+	}
+
+	private void arrayToQueue(PlayingCard[] deckCopy)
+	{
+		for (int i = 0; i < deckCopy.length; i++)
+		{
+			deck.add(deckCopy[i]);
+		}
+		
+	}
+
+	// Generates Card for every Rank and Colour and adds it in the deck
+	private void generateDeck()

```

Comment:
@Kristiyan-Ivanov 
Void methods usually mean that they have side effects. Try to minimize the side effect functions in your code. The name "generateDeck" applies that it will create deck and will return it. 
On other hand the shuffle functions is more suited for public method which changes the state of the object. If you implement the methods this way the constructor will look like this:

private void newDeck()
 +	{
 +		deck = generateDeck();
 +		shuffle();
 +	}